### ISSUE TITLE
Bug of calculating Classification Performance metrics in PerfMetricClassif

### ISSUE THREAD

**Issue by xinyuejohn** [marrlab/DomainLab#112]
Hi, I found some issues in: domainlab/utils/perf_metrics.py

Take calculating Accuracy as an example:

1. Accuracy is not reset after each calculation. In `cal_metrics()` function, every time Accuracy is updated using `update(preds, target)` and finally calculated by `compute()`. 

	`cal_metrics()` has been called twice in `cal_perf_metric` function. But for the second time, Accuracy is calculated on the inputs from first call and the second call.

2. Inputs (predictions and targets) to the classification metrics functions may not be correct.
	DomainLab is using torchmetrics==0.10.3 and input types are defined here: 	https://torchmetrics.readthedocs.io/en/v0.10.3/pages/classification.html#input-types
	
 	```
	self.acc = Accuracy(num_classes=num_classes, average=average)
	... 
	...
	with torch.no_grad():
        for i, (x_s, y_s, *_) in enumerate(loader_te):
            x_s, y_s = x_s.to(device), y_s.to(device)
            pred_label, prob, _, *_ = model_local.infer_y_vpicn(x_s)
            _, target_label = torch.max(y_s, 1)
            self.acc.update(pred_label, y_s.int())
	```
	When I tried multi-class classification (num_classes = 3):
		

	- 	shape of pred_label: (N * 3) with one-hot encoding binary label

	- 	shape of y_s: (N * 3) with one-hot encoding binary label

	
	But according to the above torchmetrics documentation, for Multi-class label input, preds shape and target shape should all be 				(N,). And I found this wrong input leads to the wrong ConfusionMatrix results.

**Comment by xinyuejohn** [comment]
@smilesun Hi, this issue occurred when I run multi-class task on our benchmark dataset. I have already spent really a long time on this issue. So I think I will ignore this and focus on benchmarking settings for now. And when the benchmark are all settled, I will try to fix this.

**Comment by smilesun** [comment]
@xinyuejohn , thanks for the efforts!  Sure, I guess we have to set up the benchmark first, I will see if i could resolve the issue

**Comment by smilesun** [comment]
resolved at https://github.com/marrlab/DomainLab/pull/113

**Comment by smilesun** [comment]
Fixed in [a3bec6f](https://github.com/marrlab/DomainLab/commit/a3bec6f67a44a97f226891e97fcb3da629134a05)

**Comment by smilesun** [comment]
@xinyuejohn, we might want to think this again, i reopened this issue due to its relation to https://github.com/marrlab/DomainLab/issues/145

**Comment by xinyuejohn** [comment]
> @xinyuejohn, we might want to think this again, i reopened this issue due to its relation to #145

I will check this again after submitting our paper.

**Comment by smilesun** [comment]
I forgot to close this issue, is is fixed already at 
`cd74075e30e3251c7b6d62b02d69b9ce4220c0e4`
